components/notifier/app/controllers/notifier/application_controller.rb

```diff
@@ -15,5 +15,14 @@ def bad_token_due_to_session_expired
         format.json { render json: { :token_expired => root_url }, status: :unauthorized }
       end
     end
+
+    def user_not_authorized(_exception)
+      flash[:error] = "You are not authorized to perform this action."
```

Comment:
Over all changes looks good but I have a minor suggestion. I believe it might be beneficial to start using translations for such messages. I understand that we haven’t used translations in this engine before for controllers but it would be a good start from here on, I appreciate your thoughts on this.